nestkernel/conn_builder_conngen.cpp

```diff
@@ -20,158 +20,140 @@
+    {
+      throw BadProperty( "A params_map has to be given if the ConnectionGenerator has values." );
+    }
+
+    updateValue< DictionaryDatum >( conn_spec, "params_map", params_map_ );
+
+    for ( Dictionary::iterator it = params_map_->begin(); it != params_map_->end(); ++it )
+    {
+      it->second.set_access_flag();
+    }
+
+    if ( syn_specs[ 0 ]->known( names::weight ) or syn_specs[ 0 ]->known( names::delay ) )
+    {
+      throw BadProperty( "Properties weight and delay cannot be specified in syn_spec if the "
+			 "ConnectionGenerator has values." );
+    }
   }
-  const index synmodel_id = static_cast< index >( synmodel );
-  DictionaryDatum dummy_params = new Dictionary();
+}
+
+void
+ConnectionGeneratorBuilder::connect_()
+{
+  cg_set_masks();
+  cg_->start();
 
-  cg_set_masks( cg, source_node_ids, target_node_ids );
-  cg->start();
+  librandom::RngPtr rng = kernel().rng_manager.get_grng();
 
   int source;
   int target;
-  const int num_parameters = cg->arity();
+  const int num_parameters = cg_->arity();
   if ( num_parameters == 0 )
   {
     // connect source to target
-    while ( cg->next( source, target, NULL ) )
+    while ( cg_->next( source, target, NULL ) )
     {
       // No need to check for locality of the target, as the mask
-      // created by cg_set_masks() only contain local nodes.
-      kernel().connection_manager.connect(
-        ( *source_node_ids )[ source ], ( *target_node_ids )[ target ], dummy_params, synmodel_id );
+      // created by cg_set_masks() only contains local nodes.
+      Node* const target_node = kernel().node_manager.get_node_or_proxy( ( *targets_ )[ target ] );
+      const thread target_thread = target_node->get_thread();
+      single_connect_( ( *sources_ )[ source ], *target_node, target_thread, rng );
     }
   }
   else if ( num_parameters == 2 )
   {
-    if ( not params_map->known( names::weight ) or not params_map->known( names::delay ) )
+    if ( not params_map_->known( names::weight ) or not params_map_->known( names::delay ) )
     {
       throw BadProperty( "The parameter map has to contain the indices of weight and delay." );
     }
 
-    const size_t w_idx = ( *params_map )[ names::weight ];
-    const size_t d_idx = ( *params_map )[ names::delay ];
+    const size_t d_idx = ( *params_map_ )[ names::delay ];
+    const size_t w_idx = ( *params_map_ )[ names::weight ];
 
-    const bool w_idx_is_0_or_1 = ( w_idx == 0 ) or ( w_idx == 1 );
     const bool d_idx_is_0_or_1 = ( d_idx == 0 ) or ( d_idx == 1 );
+    const bool w_idx_is_0_or_1 = ( w_idx == 0 ) or ( w_idx == 1 );
     const bool indices_differ = ( w_idx != d_idx );
-    if ( not( w_idx_is_0_or_1 and d_idx_is_0_or_1 and indices_differ ) )
+    if ( not( d_idx_is_0_or_1 and w_idx_is_0_or_1 and indices_differ ) )
     {
-      throw BadProperty( "w_idx and d_idx have to differ and be either 0 or 1." );
+      throw BadProperty( "d_idx and w_idx have to be either 0 or 1 and cannot be the same." );
```

Comment:
It might be better to use `delay index and weight index` instead of `d_idx and w_idx` in the message.

Reply:
What about
```suggestion
      throw BadProperty( "The indices for weight and delay have to be either 0 or 1 and cannot be the same." );
```

Reply:
Looks good!